python/test/unit/fem/test_discrete_operators.py

```diff
@@ -80,113 +78,15 @@ def test_gradient_interpolation(cell_type, p, q):
 
     # Compute global matrix vector product
     w = Function(W)
-    G.mult(u.vector, w.vector)
-    w.x.scatter_forward()
-
-    atol = 100 * np.finfo(default_real_type).resolution
-    assert np.allclose(w_expr.x.array, w.x.array, atol=atol)
-    G.destroy()
-
-
-@pytest.mark.parametrize("p", range(1, 4))
-@pytest.mark.parametrize("q", range(1, 4))
-@pytest.mark.parametrize("from_lagrange",
-                         [
-                             True,
-                             False
-                         ])
-@pytest.mark.parametrize("cell_type", [
-    CellType.quadrilateral,
-    CellType.triangle,
-    CellType.tetrahedron,
-    CellType.hexahedron
-])
-def test_interpolation_matrix(cell_type, p, q, from_lagrange):
-    """Test that discrete interpolation matrix yields the same result as interpolation."""
-    comm = MPI.COMM_WORLD
-    if cell_type == CellType.triangle:
-        mesh = create_unit_square(comm, 7, 5, ghost_mode=GhostMode.none, cell_type=cell_type)
-        lagrange = "Lagrange" if from_lagrange else "DG"
-        nedelec = "Nedelec 1st kind H(curl)"
-    elif cell_type == CellType.quadrilateral:
-        mesh = create_unit_square(comm, 11, 6, ghost_mode=GhostMode.none, cell_type=cell_type)
-        lagrange = "Q" if from_lagrange else "DQ"
-        nedelec = "RTCE"
-    elif cell_type == CellType.hexahedron:
-        mesh = create_unit_cube(comm, 3, 2, 1, ghost_mode=GhostMode.none, cell_type=cell_type)
-        lagrange = "Q" if from_lagrange else "DQ"
-        nedelec = "NCE"
-    elif cell_type == CellType.tetrahedron:
-        mesh = create_unit_cube(comm, 3, 2, 2, ghost_mode=GhostMode.none, cell_type=cell_type)
-        lagrange = "Lagrange" if from_lagrange else "DG"
-        nedelec = "Nedelec 1st kind H(curl)"
-    v_el = element(lagrange, mesh.basix_cell(), p, rank=1)
-    s_el = element(nedelec, mesh.basix_cell(), q)
-    if from_lagrange:
-        el0 = v_el
-        el1 = s_el
-    else:
-        el0 = s_el
-        el1 = v_el
-
-    V = FunctionSpace(mesh, el0)
-    W = FunctionSpace(mesh, el1)
-    G = interpolation_matrix(V._cpp_object, W._cpp_object)
-    G.assemble()
 
-    u = Function(V)
+    # Get the local part of G (no ghost rows)
```

Comment:
Is this correct? It is different from https://github.com/FEniCS/dolfinx/blob/6034c266dea5185eca26e1cedd226de1f18adfdf/python/test/unit/fem/test_expression.py#L124.

Reply:
I don't see much of a difference - except for explicitly telling scipy the shape - but it can infer that.